internal/payloads/reservation_payload.go

```diff
@@ -2,33 +2,58 @@ package payloads
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/RHEnVision/provisioning-backend/internal/models"
 	"github.com/go-chi/render"
 )
 
 // ReservationRequest is empty, account comes in HTTP header and
 // provider type in HTTP URL. All other fields are auto-generated.
-type ReservationRequest struct {
-}
 
-type ReservationResponse struct {
-	*models.Reservation
-}
+type ReservationPayload struct {
+	ID int64 `json:"id"`
 
-type NoopReservationRequest ReservationRequest
+	// Provider type. Required.
+	Provider int `json:"provider"`
 
-type NoopReservationResponse struct {
-	*models.NoopReservation
+	// Account ID. Required.
+	AccountID int64 `json:"-"`
+
+	// Time when reservation was made.
+	CreatedAt time.Time `json:"created_at"`
+
+	// Textual status of the reservation or error when there was a failure
+	Status string `json:"status"`
+
+	// Time when reservation was finished or nil when it's still processing.
+	FinishedAt time.Time `json:"finished_at"`
+
+	// Flag indicating success, error or unknown state (NULL). See Status for the actual error.
+	Success bool `json:"success"`
 }
 
-type AWSReservationRequest struct {
+type AWSReservationPayload struct {
+	ReservationPayload
 	*models.AWSReservation
 }
 
-type AWSReservationResponse struct {
-	*models.AWSReservation
+type NoopReservationPayload struct {
+	ReservationPayload
+	*models.NoopReservation
 }
+type ReservationRequest struct {
+}
+
+type ReservationResponse ReservationPayload
+
+type NoopReservationRequest ReservationRequest
+
+type NoopReservationResponse NoopReservationPayload
+
+type AWSReservationRequest AWSReservationPayload
+
+type AWSReservationResponse AWSReservationPayload
```

Comment:
I think this gives us opportunity to split request and response payloads and customize them a bit. I can imagine that for both types, requests do not need the following fields to be present:

* AccountID (this one can be completely deleted from all payloads)
* ID (because this is generated on save)
* Provider (because that is part of the URL)
* Created
* Status
* Finished
* Success (all the four are relevant only for response)

This all applies to both AWS and Noop. For the latter, it will probably be a completely empty request payload. We want to remove them, because then they will not be generated in the OpenAPI and it will not confuse API users.

Responses are a different story, there we need all of the above. As I said, NULLable fields must be pointers tho.